### PR TITLE
Sanitize the setup of the default Ide.Config

### PR DESCRIPTION
**Pull request by pepeiborra**
This adds a field in the `defaultMain` driver to specify the default config, replacing two callbacks and making things a bit saner. 

The rest of the changes are to deal with the use of `def` in various places which had to be turned into a parameter.

### CODE REVIEW

**Review by Ailrun — APPROVED** [review]
LGTM

**Comment by Ailrun** [comment]
It looks like HLint plugin requires some changes too.

**Comment by pepeiborra** [comment]
> It looks like HLint plugin requires some changes too.

there isn't a sensible way to fix it, so I just plugged in the type error and let the lsp 1.0 PR take care of it

File: plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs [hunk 1]
```diff
@@ -102,7 +103,7 @@ type instance RuleResult GetHlintDiagnostics = ()
 rules :: PluginId -> Rules ()
 rules plugin = do
   define $ \GetHlintDiagnostics file -> do
-    config <- getClientConfigAction
+    config <- getClientConfigAction def
```
Comment (Ailrun) [inline plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs:106]: We also need to update the dependency for this package too

**Comment by pepeiborra** [comment]
@jneira I'm dropping the "ghcide does not support update config" test. Shout if you want to keep it

**Comment by wz1000** [comment]
@pepeiborra that test is already dropped in my lsp-1.0 PR.

**Comment by jneira** [comment]
I will not miss it, thanks for cleaning it out

**Comment by pepeiborra** [comment]
This keeps failing with: 

image

**Pull request merged**